blueocean-dashboard/src/main/js/util/UrlUtils.js

```diff
@@ -140,10 +140,90 @@ export function calculateRunLogURLObject(config) {
 }
 
 /**
+ * Provide a pagination function for the generic
+ * blueocean pagination
+ */
+export function paginateUrl(url) {
+    const sep = url.indexOf('?') >= 0 ? '&' : '?';
+    return (start, limit) => `${url}${sep}start=${start}&limit=${limit}`;
+}
+
+/**
+ * Examines the provided object for:
+ * organization, pipeline, branch, runId
+ * and builds a path to the thing as best it can...
+ */
+export function getRestUrl({ organization, pipeline, branch, runId }) {
+    const pipelineName = typeof pipeline === 'object' ? pipeline.fullName : pipeline;
+    const organizationName = organization ||
+        (typeof pipeline === 'object' ? pipeline.organization : '');
+    const jenkinsUrl = require('../config').getJenkinsRootURL();
+    let url = `${jenkinsUrl}/blue/rest/organizations/${encodeURIComponent(organizationName)}`;
+    if (pipelineName) {
+        // pipelineName might include a folder path, don't escape it
+        url += `/pipelines/${pipelineName}`;
+    }
+    if (branch) {
+        url += `/branches/${encodeURIComponent(branch)}`;
+    }
+    if (runId) {
+        url += `/runs/${encodeURIComponent(runId)}`;
+    }
+    return url;
+}
+
+/**
+ * Examines the provided object for:
+ * organization, pipeline, branch, runId
+ * and builds a path to the GUI as best it can...
+ * TODO: we could probably figure this out more accurately from the routes
+ */
+export function getLocation({ location, organization, pipeline, branch, runId, tab, basePage }) {
+    const pipelineName = typeof pipeline === 'object' ? pipeline.fullName : pipeline;
+    const organizationName = organization ||
+        (typeof pipeline === 'object' ? pipeline.organization : '');
+    const basePageName = basePage || location &&
+        (location.pathname ? location.pathname : location).split('/')[4];
+    /* eslint-disable prefer-template */
+    let url = '/organizations/' + encodeURIComponent(organizationName);
+    if (pipelineName) {
+        url += '/' + encodeURIComponent(pipelineName);
+    }
+    if (basePageName) {
+        url += '/' + basePageName;
+    }
+    if (runId) {
+        if (!basePageName) {
+            url += '/activity';
+        }
+        url += '/detail' +
+        (branch ? ('/branch/' + uriString(branch)) : '') +
+        '/' + encodeURIComponent(runId) +
+        (tab ? ('/' + tab) : '');
+    }
+    /* eslint-enable prefer-template */
+    return url;
+}
```

Comment:
Note, the ability to click on a Favorite card in main Dashboard is broken since the URL structure to run details has changed. It'd be good if we could promote this code so that it could be shared between the plugins. I started the Commons JS project in #404 - we could possibly get that merged to master, then refactor this func into the shared lib? Otherwise maybe just dup the code in "blueocean-personalization" temporarily.